language-support/hs/bindings/src/DA/Ledger/LedgerService.hs

```diff
@@ -4,27 +4,46 @@
 -- Abstraction for LedgerService, which can be composed monadically.
 module DA.Ledger.LedgerService (
     LedgerService, runLedgerService, makeLedgerService, TimeoutSeconds,
+    Jwt, setToken,
     askTimeout,
     ) where
 
 import Control.Monad.Fail (MonadFail)
 import Control.Monad.IO.Class (MonadIO)
-import Control.Monad.Trans.Reader (ReaderT(..),runReaderT,ask)
+import Control.Monad.Trans.Reader (ReaderT(..),runReaderT,ask,local)
+import DA.Ledger.Jwt (Jwt)
 import DA.Ledger.Retry (ledgerRetry)
 import Network.GRPC.HighLevel.Client(TimeoutSeconds)
-import Network.GRPC.HighLevel.Generated(ClientConfig)
+import Network.GRPC.HighLevel.Generated(ClientConfig,MetadataMap(..))
 import UnliftIO(MonadUnliftIO)
+import qualified DA.Ledger.Jwt as Jwt (toString)
+import qualified Data.ByteString.UTF8 as BSU8
+import qualified Data.Map as Map
+import qualified Data.SortedList as SortedList
 
-type Context = (TimeoutSeconds,ClientConfig)
+type Context = (TimeoutSeconds,ClientConfig,Maybe Jwt)
```

Comment:
I think it would be nicer for this to be a record instead of a type synonym for a tuple.